wolfcrypt/src/port/maxim/maxq10xx.c

```diff
@@ -0,0 +1,3376 @@
+        }
+        else {
+            rc = BAD_FUNC_ARG;
+        }
+    }
+#endif /* WOLFSSL_MAXQ108X */
+
+    if (rc != 0 && rc != CRYPTOCB_UNAVAILABLE) {
+        rc = WC_HW_E;
+    }
+
+    return rc;
+}
+
+static int wolfSSL_Soft_CryptoDevCb(int devId, wc_CryptoInfo* info, void* ctx)
+{
+    (void)devId;
+    (void)info;
+    (void)ctx;
+
+    return CRYPTOCB_UNAVAILABLE;
+}
+#endif /* WOLF_CRYPTO_CB */
+
+#ifdef WOLFSSL_MAXQ10XX_TLS
+#ifdef WOLFSSL_MAXQ108X
+static int calculate_modulus_offset(const unsigned char * cert_data, int offset)
+{
+    int i;
+    int l_offset = offset;
+    for (i = 0; i < 2; i++) {
+        if ((cert_data[l_offset]) & 0x80) {
+            if ((cert_data[l_offset] & 0x7f) == 1) {
+                l_offset += 3;
+            }
+            if ((cert_data[l_offset] & 0x7f) == 2) {
+                l_offset += 4;
+            }
+        }
+        else {
+            l_offset += 2;
+        }
+    }
+    return l_offset;
+}
+
+static int maxq_curve_wolfssl_id2mxq_id(word32 curve_id, mxq_length *keycomplen)
+{
+    switch (curve_id) {
+    case ECC_SECP256R1_OID:
+        *keycomplen = 32;
+        return MXQ_KEYPARAM_EC_P256R1;
+    case ECC_SECP384R1_OID:
+        *keycomplen = 48;
+        return MXQ_KEYPARAM_EC_P384R1;
+    case ECC_SECP521R1_OID:
+        *keycomplen = 66;
+        return MXQ_KEYPARAM_EC_P521R1;
+    case ECC_BRAINPOOLP256R1_OID:
+        *keycomplen = 32;
+        return MXQ_KEYPARAM_EC_BP256R1;
+    case ECC_BRAINPOOLP384R1_OID:
+        *keycomplen = 48;
+        return MXQ_KEYPARAM_EC_BP384R1;
+    case ECC_BRAINPOOLP512R1_OID:
+        *keycomplen = 64;
+        return MXQ_KEYPARAM_EC_BP512R1;
+    default:
+        return MXQ_UNKNOWN_CURVE;
+    }
+}
+
+static int mxq_get_sign_alg_from_sig_oid(word32 maxq_id) {
+
+    switch (maxq_id) {
+    case CTC_SHA256wECDSA:
+        return ALGO_ECDSA_SHA_256;
+    case CTC_SHA384wECDSA:
+        return ALGO_ECDSA_SHA_384;
+    case CTC_SHA512wECDSA:
+        return ALGO_ECDSA_SHA_512;
+    case CTC_SHA256wRSA:
+        return ALGO_RSASSAPSSPKCS1_V1_5_SHA256;
+    case CTC_SHA384wRSA:
+        return ALGO_RSASSAPSSPKCS1_V1_5_SHA384;
+    case CTC_SHA512wRSA:
+        return ALGO_RSASSAPSSPKCS1_V1_5_SHA512;
+    default:
+        return ALGO_INVALID;
+    }
+}
+#endif /* WOLFSSL_MAXQ108X */
+
+/* TODO: Get rid of publicKeyIndex. Pass in the information it is parsing.
+ *       Then Improve this to pass in the the parts of decoded cert it needs.
+ */
+static int maxq10xx_process_server_certificate(WOLFSSL* ssl,
+                                               DecodedCert* p_cert)
+{
+    (void)ssl;
+    mxq_keytype_id_t key_type = MXQ_KEYTYPE_ECC;
+    mxq_keyparam_id_t keyparam = MXQ_KEYPARAM_EC_P256R1;
+    mxq_length totalkeylen;
+    mxq_algo_id_t sign_algo = ALGO_ECDSA_SHA_256;
+    int pk_offset = p_cert->publicKeyIndex;
+    mxq_length keycomplen = 32;
+    int rc;
+    mxq_err_t mxq_rc;
+    mxq_u1 certdata[2048];
```

Comment:
Size could be less for ECC only. Do not like hard coded values. Make macro and put at top based on algos.

Reply:
> Size could be less for ECC only. 

Maybe?  The data that goes in here is opaque.  its produced by `MXQ_Build_EC_Cert()` (Ignore EC in function name) and then passed to MXQ_ImportChildCert() so we shouldn't really change it based on algorithm.  I've defined `MAX_CERT_DATASIZE` and used that.